pghoard/pghoard.py

```diff
@@ -609,6 +621,29 @@ def startup_walk_for_missed_files(self):
 
                 filetype = FileType.Timeline if wal.TIMELINE_RE.match(filename) else FileType.Wal
 
+                # verify if file was already compressed, otherwise the transfer agent will encounter
+                # duplicated UploadEvents. In case it was compressed, we should just add it to the deletion queue
+                base_compressed_file_path = (
+                    compressed_timeline_path if filetype is FileType.Timeline else compressed_xlog_path
+                )
+                is_already_compressed = os.path.exists(os.path.join(base_compressed_file_path, filename))
+                has_metadata_file = os.path.exists(os.path.join(base_compressed_file_path, filename + ".metadata"))
+
+                # the file was compressed correctly
+                if is_already_compressed and has_metadata_file:
+                    self.log.debug("Uncompressed file %r is already compressed, adding to deletion queue.", full_path)
+                    if filetype is FileType.Timeline:
+                        os.unlink(full_path)
+                    else:
+                        delete_request = WalFileDeletionEvent(backup_site_name=site, file_path=Path(full_path))
+                        self.wal_file_deletion_queue.put(delete_request)
```

Comment:
We should probably extend `WALFileDeleterThread` to remove all files without leaving the last one for the events coming from here, because otherwise we will hit the issue of trying to delete twice again.